src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -509,6 +516,19 @@ private void ParseHelper(string[] args)
                     switchKey = switchKey.Substring(1);
                 }
 
+                // If version is in the commandline, don't continue to look at any other parameters
```

Comment:
Furthermore we can always do something like `powershell -version:2` if people really demand that functionality in the future (and actually @lzybkr and @BrucePay agreed that it would even be acceptable to do `powershell -version 2` as it's highly unlikely that anyone wants to echo one int when they shell out to PowerShell).